src/provider.js

```diff
@@ -160,16 +160,25 @@ Provider.prototype.getInterface = function () {
   if (this.iface) {
     return this.iface;
   } else {
+    var sanityCheck = function (prov) {
```

Comment:
right. the third commit here is the logger, and is built on top of the other branch.